niimpy/preprocess.py

```diff
@@ -1099,21 +1099,23 @@ def location_data(database,subject,begin=None,end=None):
 
 #Screen
 
-def get_battery_data(database, user, start = None, end = None):
+def get_battery_data(battery, user=None, start = None, end = None):
     """ Returns a DataFrame with battery data for a user.
-
     Parameters
     ----------
-    database: Niimpy database
-    user: string
+    battery: DataFrame with battery data
+    user: string, optional
     start: datetime, optional
     end: datetime, optional
     """
-    assert isinstance(database, niimpy.database.Data1),"database not given in Niimpy database format"
-    assert isinstance(user, str),"user not given in string format"
-
-    battery_data = database.raw(table='AwareBattery',user=user)
-
+    assert isinstance(battery, pd.core.frame.DataFrame), "data is not a pandas DataFrame"
+    
+    if(user!= None):
+        assert isinstance(user, str),"user not given in string format"
+        battery_data = battery[(battery['user']==user)]
+    else:
+        battery_data = battery
```

Comment:
I would recommend doing this like we did in screen.py:

```python
    screen  = read._get_dataframe(screen, table='AwareScreen', user=subject)
    screen  = niimpy.filter_dataframe(screen, begin=begin, end=end)
```

- It shouldn't limit to dataframes only, the first line handles database AND dataframes backwards-compatible
- user doesn't have to only be string format (incoming PR)
- The other filtering (following lines) happens within the filter function.

This makes the minimal code with least copy-and-paste, and is easy to improve later on.